source/admin_manual/list_of_events.rst

```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
-| uid                 | Mail IMAP UID number                                 |
-+---------------------+------------------------------------------------------+
+.. dovecot_event:field_group:: mail
 
-.. _event_mail_opened:
+   :field seq: Mail sequence number.
+   :field uid: Mail IMAP UID number.
 
-mail_opened
-^^^^^^^^^^^
 
-.. versionadded:: 2.3.15
+.. dovecot_core:event:: mail_opened
+   :added: v2.3.15
+   :inherit: mail, mailbox
 
-A mail was opened e.g. for reading its body. Note that this event is not sent
-when mails' metadata is accessed, even if it causes opening the mail file.
+   :field reason: Reason why the mail was opened. (optional)
 
-+---------------------+--------------------------------------------------------+
-| Field               | Description                                            |
-+=====================+========================================================+
-| reason              | Reason why the mail was opened. (optional)             |
-+---------------------+--------------------------------------------------------+
+   A mail was opened e.g. for reading its body. Note that this event is not
+   sent when mails' metadata is accessed, even if it causes opening the mail
+   file.
 
-.. _event_mail_expunge_requested:
 
-mail_expunge_requested
-^^^^^^^^^^^^^^^^^^^^^^
+.. dovecot_core:event:: mail_expunge_requested
+   :added: v2.3.15
+   :inherit: mail, mailbox
 
-.. versionadded:: 2.3.15
+   A mail is set to be expunged. (Note that expunges can be rolled back later
+   on, this event is emitted when an expunge is requested).
 
-A mail is set to be expunged. (Note that expunges can be rolled back later on,
-this event is emitted when an expunge is requested).
 
-Mail index
+Mail Index
 ==========
 
-.. _event_mail_index:
-
-Mail index
-----------
+Index
+-----
 
 Index file handling for ``dovecot.index*``, ``dovecot.map.index*``,
 ``dovecot.list.index*`` and similar indexes.
 
-+---------------------+------------------------------------------------------+
-| Field               | Description                                          |
-+=====================+======================================================+
-| Inherits from :ref:`event_mailbox`, :ref:`event_storage` or                |
-| :ref:`event_mail_user` depending on what the index is used for.            |
-+---------------------+------------------------------------------------------+
+.. todo:: mail index "Inherits from event_mailbox, event_storage or
+          event_mail_user depending on what the index is used for."
 
+.. dovecot_core:event:: mail_index_recreated
+   :added: v2.3.12
 
-.. _event_mail_index_recreated:
+   :field filepath: Path to the index file being recreated.
+   :field reason: Reason why the mail index was recreated.
 
-mail_index_recreated
-^^^^^^^^^^^^^^^^^^^^
+   A mail index file was recreated.
 
-.. versionadded:: 2.3.12
+   .. todo:: list all possible reasons
```

Comment:
The reason is human-readable string, can't list all possible reasons. Could modify to say it:

```
   :field reason: Human-readable reason why the mail index was recreated.
```